src/App.js

```diff
@@ -7,22 +7,39 @@ class App extends React.Component {
   constructor() {
     super()
     this.state = {
-      movies: movieData.movies
+      movies: [],
+      error: ''
     }
     this.showMovieDetails = this.showMovieDetails.bind(this)
     this.goBack = this.goBack.bind(this)
   }
+  fetchAllMovies() {
+    fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
+    .then(response => {
+      if(!response.ok) {
+        throw new Error('sorry try again')
+      } else {
+        return response.json()
+      }
+    })
+    .then(data => this.setState({ movies: data.movies}))
+    .catch(err => this.setState({error: 'Something went wrong with our server please try again'}))
+  }
+  componentDidMount() {
+    return this.fetchAllMovies()
+  }
   showMovieDetails = (id) => {
     const selectedMovie = this.state.movies.filter(movie => movie.id === id)
     this.setState({movies: selectedMovie})
   }
   goBack = () => {
-    this.setState({movies: movieData.movies})
+    this.fetchAllMovies()
   }
```

Comment:
I imagine this will change once you refactor to use Router, but we shouldn't need to re-fetch after the initial fetch.